src/coreclr/src/jit/lowerxarch.cpp

```diff
@@ -920,8 +920,61 @@ void Lowering::LowerFusedMultiplyAdd(GenTreeHWIntrinsic* node)
 //
 void Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
 {
+    if (node->TypeGet() == TYP_SIMD12)
+    {
+        // GT_HWINTRINSIC node requiring to produce TYP_SIMD12 in fact
+        // produces a TYP_SIMD16 result
+        node->gtType = TYP_SIMD16;
+    }
+
     switch (node->gtHWIntrinsicId)
     {
+        case NI_SSE2_CompareGreaterThan:
```

Comment:
I opted to move most of the comparison logic here to lowering.

Previously, several of these weren't table driven and had custom importation/codegen logic. However, given that they only require an operand swap and the `ival` lookup was recently made dynamic rather than table driven (which reduced the size of the info tables and was largely needed anyways due to the ival differing between VEX and non-VEX), I was able to instead just take the encoding into account here and swap the args if necessary.

This also means, for example, we can just create a `CompareLesserThan` node for the SimdAsHWIntrinsic logic, rather than also needing to take the various ISAs, encodings, etc into account there as well.